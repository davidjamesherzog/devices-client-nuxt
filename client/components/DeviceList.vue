<template>
  <div>
    <device-list-item
      v-for="device in devices"
      :key="device.id"
      :device="device"
    />
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'nuxt-property-decorator';
import Device from '~/client/models/Device';
import DeviceListItem from '~/components/DeviceListItem.vue';

@Component({
  name: 'DeviceList',
  components: {
    DeviceListItem
  }
})
export default class DeviceList extends Vue {
  @Prop({ type: Array })
  private devices!: Array<Device>;
}
</script>

<style lang="scss" scoped></style>
