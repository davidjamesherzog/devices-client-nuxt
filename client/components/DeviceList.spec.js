import { createLocalVue, mount } from '@vue/test-utils';
import Buefy from 'buefy';
import DeviceList from '~/components/DeviceList.vue';

describe('DeviceList', () => {
  describe('template', () => {
    let localVue;

    beforeEach(() => {
      localVue = createLocalVue();
      localVue.use(Buefy);
    });

    test('should render device list', () => {
      const wrapper = mount(DeviceList, {
        localVue,
        propsData: {
          devices: [
            {
              id: 1,
              name: 'Macbook-Dave',
              description: 'Macbook laptop',
              os: 'Mac'
            }
          ]
        }
      });
      const id = wrapper.find('[data-list-item-id="1"]');
      expect(id).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
