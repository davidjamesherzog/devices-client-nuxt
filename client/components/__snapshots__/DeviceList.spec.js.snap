// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceList template should render device list 1`] = `
<div>
  <div data-list-item-id="1" class="box">
    <article class="media">
      <div class="media-left">
        <div class="content has-text-centered"><span class="icon has-text-primary is-large" data-card-icon=""><i class="mdi mdi-laptop mdi-48px"></i></span></div>
      </div>
      <div class="media-content">
        <div class="content">
          <p><strong data-list-item-title="1">Macbook-Dave</strong> <small data-list-item-os="1">Mac</small> <br> <span data-list-item-description="1">
            Macbook laptop
          </span></p>
        </div>
      </div>
    </article>
  </div>
</div>
`;
